SPEC/FILESv2.md

```diff
@@ -0,0 +1,237 @@
+
+If imported files are added to MFS, we _could_ remove `pin` in it's entirety from a small focused "core". Other APIs like `config`, `bitswap`, `block`, `bootstrap`, `dag`, `dht`, `object`, `pin`, `ping`, `pubsub`, `refs` could also be removed to create an even leaner core (although in some cases aspects of these APIs would still be in use behind the scenes).
+
+This problem requires further definition and reasoning and consequently it is outside the scope of the solutions in this proposal, but it's worth entertaining nethertheless!
+
+### 7. `cat` and `files.read` are the same
+
+These methods perform the same operation, and `files.read` also already works with both MFS and IPFS paths.
+
+### 8. A single imported file loses it's name
+
+The common case of importing a single file to IPFS gives us back a CID but the file loses it's file name in the process unless the user explicitly asks for it to be "wrapped" in a directory. This is a big hurdle to overcome mentally and a significant WTF moment for people new to IPFS. This also means that the file extension is lost, losing any hint of what is inside. Also, exporting a file out of IPFS to a user's OS yields a file that the OS does not know how to open. Finally, having no name is bad for SEO both on the gateway or otherwise.
+
+---
+
+## Solutions
+
+The solutions proposed below solve the majority of the problems but not all of them. Each solution summarises the problem that is being addressed but it is strongly recommended that the [problems](#problems) section is studied for the full detail.
+
+### 1. Rename `add` to `import`
+
+"Import" more accurately describes what is occurring and will prevent confusion with `write`.
+
+```sh
+ipfs import --help
+# ...
+```
+
+It's also the [current name of the API method to import data into `go-filecoin`](https://github.com/filecoin-project/go-filecoin/blob/204837f72d20bd89889fcf92061c8846b238ccf4/cmd/go-filecoin/client.go#L62), which is reassuring.
+
+#### 1.1 Always add to MFS
+
+Adding files to MFS removes any performance overhead of creating/maintaining pinset DAG nodes, unburdens the user from understanding pinning (for the most part), improves visibility of added files and makes it significantly easier to find or remove files that were previously imported.
+
+The `ipfs import` command _optionally_ takes a MFS path option `--dest`, a directory into which imported files are placed. Note the destination directory is automatically created (but not any parents). If the destination directory exists already then an error is thrown, unless the `--overwrite` flag is set. This causes any existing files with the same name as the imported files to be overwritten.
```

Comment:
Small point, but maybe `--force` instead of `--overwrite`?  Feels a bit more unixy.